### PR TITLE
New admin branch

### PR DESCRIPTION
**Pull request by codeMonkeyHopeful**
try 2 for the choresRoutes issue

### CODE REVIEW

File: server/apiRoutes/adminRoutes.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+/*
+ * @ROUTE: POST to /api/admin/add_chore
+ * @DESC: Allows for an admin to add a chore to their specific group
+ * @ACCESS: admin only
+ */
+router.post('/add_chore', async (req, res, next) => {
+  await updateChoreStatus();
+  const { userId, name, difficulty, timeLimit, details } = req.body;
+  const userInfo = await UserGroup.findOne({ where: { userId } });
+  const detailsArray = Array(details);
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: how is details being sent to the server? Not as an array already? Why do we have to create an array from it?
in postman we can use urlencoded but in our client we would be sending json

File: server/apiRoutes/adminRoutes.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+ * @ROUTE: POST to /api/admin/extend_chore_time
+ * @DESC: Allows for an admin to extend a chores time by feeding new due date
+ * @ACCESS: admin only
+ */
+router.post('/extend_chore_time', async (req, res, next) => {
+  await updateChoreStatus();
+  const { userIdToModify, choreId, newDueDate } = req.body;
+  const adjusted = new Date(newDueDate);
+  console.log(adjusted);
+  AssignedChore.update(
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: this is going to find all assigned chores with that userId and choreId. We should be able to query by assignedChoreId (where:{id: assignedChoreId})

File: server/apiRoutes/adminRoutes.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    { expiresOn: adjusted },
+    { where: { userId: userIdToModify, choreId } }
+  )
+    .then(newChore => res.send(newChore))
+    .catch(next);
+});
+
+/*
+ * @ROUTE: POST to /api/admin/add_new_user
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js:172]: description is off

File: server/apiRoutes/adminRoutes.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+ * @DESC: Allows for an admin to extend a chores time
+ * @ACCESS: admin only
+ */
+router.post('/add_new_user', async (req, res, next) => {
+  await updateChoreStatus();
+  const { userId, firstName, surName, email, groupId, adminRights } = req.body;
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: user will be created when they log in. we are simply adding them to the usergroup model

File: server/apiRoutes/adminRoutes.js [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    users.push(groupUsers[i].userId);
+  }
+
+  //Pull out all chores that are pending
+  const pendingChores = await AssignedChore.findAll({
+    where: { status: 'pending', userId: { [Op.in]: users } },
+  });
+  const pendingArr = [];
+  for (let i = 0; i < pendingChores.length; i++) {
+    pendingArr.push(pendingChores[i].id);
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: pendingChores[i].choreId?

File: server/apiRoutes/adminRoutes.js [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+/*
+ * @ROUTE: POST to /api/admin/assign_chores
+ * @DESC: Route that will randomly assign all chores to the admins group
+ * @ACCESS: admin only
+ */
+router.post('/assign_chores', async (req, res, next) => {
+  const { userId } = req.body;
+  await updateChoreStatus();
+  const allChores = await Chore.findAll({});
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js:215]: needs a groupId

File: server/apiRoutes/adminRoutes.js [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    if (!pendingArr.includes(allChores[i].id)) {
+      choresToAssign.push(allChores[i].id);
+    }
+  }
+  //users and chores to assign
+  //stir it up ... little darling ... stir it up
+  shuffleArray(users);
+  shuffleArray(choresToAssign);
+
+  if (userIsAdmin) {
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: you should check if user is admin at the very beginning all by itself
if(!userIsAdmin){
  return res.sendStatus(400)
}

File: server/apiRoutes/adminRoutes.js [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+      const userId = users[userIndex];
+      await AssignedChore.create({
+        status: 'pending',
+        userId,
+        choreId: choresToAssign[i],
+      });
+      userIndex += 1;
+    }
+
+    res.send('Success!');
```
Comment (HypeDis) [inline server/apiRoutes/adminRoutes.js]: dont forget your status codes! also, send({message:'success'}) or something like that. easier to parse in the future if you need to attach other data to the response

**Review by HypeDis — CHANGES_REQUESTED** [review]

File: server/apiRoutes/adminRoutes.js (hunk 1 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js]: updated to account for both array and non-array

File: server/apiRoutes/adminRoutes.js (hunk 2 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js]: updated to take in and find a chore by the id fed in

File: server/apiRoutes/adminRoutes.js (hunk 3 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js:172]: fixed

File: server/apiRoutes/adminRoutes.js (hunk 4 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js]: updated logic to allow for user to be assigned to groups if the assigner is an admin.  Also updated with status codes

File: server/apiRoutes/adminRoutes.js (hunk 5 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js]: pulls the chores id to bump up against the list of all chores and pull that information in for assignment

File: server/apiRoutes/adminRoutes.js (hunk 8 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/adminRoutes.js]: updated

**Review by HypeDis — APPROVED** [review]

**Pull request merged**